src/wayland/compositor/mod.rs

```diff
@@ -564,18 +564,43 @@ impl CompositorClientState {
     }
 }
 
-#[doc(hidden)]
 impl CompositorState {
-    /// Create new [`wl_compositor`](wayland_server::protocol::wl_compositor)
-    /// and [`wl_subcompositor`](wayland_server::protocol::wl_subcompositor) globals.
+    /// Create new [`wl_compositor`] version 5 and [`wl_subcompositor`] globals.
     ///
     /// It returns the two global handles, in case you wish to remove these globals from
     /// the event loop in the future.
+    ///
+    /// [`wl_compositor`]: wayland_server::protocol::wl_compositor
+    /// [`wl_subcompositor`]: wayland_server::protocol::wl_subcompositor
     pub fn new<D>(display: &DisplayHandle) -> Self
     where
         D: GlobalDispatch<WlCompositor, ()> + GlobalDispatch<WlSubcompositor, ()> + 'static,
     {
-        let compositor = display.create_global::<D, WlCompositor, ()>(5, ());
+        Self::new_with_version::<D>(display, 5)
+    }
+
+    /// The same as [`new`], but binds at least version 6 of [`wl_compositor`].
+    ///
+    /// This means that for clients to scale and apply transformation with
+    /// non-default values [`wl_surface::preferred_buffer_scale`] and
+    /// [`wl_surface::preferred_buffer_transform`] must be sent.
+    ///
+    /// [`new`]: Self::new
+    /// [`wl_compositor`]: wayland_server::protocol::wl_compositor
+    /// [`wl_surface::preferred_buffer_scale`]: WlSurface::preferred_buffer_scale
+    /// [`wl_surface::preferred_buffer_transform`]: WlSurface::preferred_buffer_transform
```

Comment:
No sorry, we are not asking downstream to send raw events. Downstream would have to check the version of global the client has bound to do this correctly, etc. This needs a function similar to `set_preferred_scale` for the fractional scale protocol.